config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleEntityBuilder.java

```diff
@@ -43,25 +46,31 @@
     }
 
     public Map<String, BundleArtifacts> build(Bundle bundle, EntityBuilder.BundleType bundleType,
-                                               Document document, String projectName,
-                                               String projectGroupName, String projectVersion) {
+                                              Document document, String projectName,
+                                              String projectGroupName, String projectVersion) {
 
         Map<String, BundleArtifacts> artifacts = buildAnnotatedEntities(bundleType, bundle, document, projectName,
                 projectGroupName, projectVersion);
         if (artifacts.isEmpty()) {
             List<Entity> entities = new ArrayList<>();
             entityBuilders.forEach(builder -> entities.addAll(builder.build(bundle, bundleType, document)));
             final Element fullBundle = bundleDocumentBuilder.build(document, entities);
+            // Create DELETE bundle - ALWAYS skip environment entities
+            final Element deleteBundleElement = createDeleteBundle(document, entities, bundle,
+                    null);
+
+            // Create DELETE Environment bundle
+            final Element deleteEnvBundleElement = createDeleteEnvBundle(document, entities);
```

Comment:
createDeleteEnvBundle gets executed two times if we call build-environment-bundle task

Reply:
Imagine we have metadata in un-annotated entities case

Reply:
@graju256 , do we have to generate environment bundle only when build-environment-bundle task is executed ?